### ISSUE TITLE
Error handling

### ISSUE THREAD

**Issue by sionovd** [ItamarWilf/PipeRT#28]
We need to add error handling and recovery. 
This is relevant for situations where something goes wrong with the connection to the video stream, or with the connection to the redis server. (If anyone can think of additional possible scenarios, please share in the comments)

The easy and naive way to do this would be to add an infinite loop which tries to re-do whichever task has failed. But it would be preferable to add the ability to shut down the component/whole pipeline after a certain period of time (if this is case, then we need to finish  #27 first)

**Comment by MayoG** [comment]
With the PipelineManager you are able to stop and start any component whenever you need, about the errors that you mentioned, I think that every routine individually need to decide what to do on this cases.
Maybe we need to let the user know what is happening in each routine using its logger and add this type of mechanism/method to the PipelineManager.

**Comment by sionovd** [comment]
I agree with both points - I think the routine should report the error and also try to handle it. If it reaches a certain configurable timeout (meaning that it can't handle the error) then it should delegate the handling to the PipelineManager.

I'd be happy to handle this issue once the PipelineManager is done :)
